test/utilities/notifications.ts

```diff
@@ -61,6 +62,73 @@ export async function notificationIsPresent(
   return false;
 }
 
+export async function notificationIsPresentWithTimeout(
+  workbench: Workbench,
+  notificationMessage: string,
+  durationInSeconds: number,
+  matchExactString: boolean = true
+): Promise<boolean> {
+
+  // Change timeout from seconds to milliseconds
+  durationInSeconds *= 1000;
+
+  const startDate = new Date();
+
+  // Keep on searching for the notification until it is found or the timeout is reached
+  while (true) {
```

Comment:
I think we should have like an actual condition other than true. Like the timeout or something